ingress/src/main/java/com/redhat/developer/ingress/IngressServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.redhat.developer.ingress;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.jboss.logging.Logger;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.redhat.developer.infra.utils.CloudEventUtils;
+import com.redhat.developer.ingress.producer.KafkaEventPublisher;
+
+import io.cloudevents.CloudEvent;
+
+@ApplicationScoped
+public class IngressServiceImpl implements IngressService {
+    private final Logger LOGGER = Logger.getLogger(IngressServiceImpl.class);
+
+    private List<String> deployments = new ArrayList<>();
+
+    @Inject
+    KafkaEventPublisher kafkaEventPublisher;
+
+    @Override
+    public boolean processEvent(String name, JsonNode event) {
+        //TODO: remove after we move to k8s
+        if (!deployments.contains(name)) {
+            throw new RuntimeException("Ingress with name " + name + " is not deployed.");
+        }
+
+        CloudEvent cloudEvent = CloudEventUtils.build(UUID.randomUUID().toString(), name,
```

Comment:
Actually the idea was to wrap the ingress event with our cloud event so to add more information about the event itself useful for our purposes (for example when it was inserted etc.. ). I've implemented the requirement that ALSO the ingress event must be a cloud event. 
If you think we should just take the event and send it to kafka directly let me know @robpblake

Reply:
@r00ta So the original `CloudEvent` is set as the `data` field on the a new `CloudEvent` before being sent to Kafka? Have I followed correctly?

Reply:
Yes, correct

Reply:
OK, thanks for the explanation 👍 

I like your idea of adding metadata related to our handling to the received Event. I am not 100% sure on the idea of putting the original event into the `data` field of a new CloudEvent as I wonder if it makes our filtering logic a bit more complex later on? I also wonder if it will make things easier for us to preserve the original `source`, `type` and `id` on the CloudEvent. 

The other approach for metadata I'm thinking about is that we add a set of namespaced attributes to the received CloudEvent e.g:

- com.redhat.eventthing/time-received
- com.redhat.eventthing/another-useful-attribute

This would leave the original CloudEvent intact.

My instinct right now is to just send the `CloudEvent` as-is onto Kafka and then we can revisit that decision once we get further into the implementation.

WDYT?

Reply:
My idea was to use the cloud event envelop just for the transport layer. The `executor` would then extract the user/incoming/real cloud event and apply the logic there. 

I'm pretty sure we will have to add some metadata to the event sooner or later, and we can achieve that wrapping the original event (like in this proposal) or modifying the original event adding some metadata there. 
Personally, I prefer the first option like I did in this PR, but I'm fine to postpone it when we will have the need (in case). 

I'll change the implementation so to send the original event as it is to kafka